akka-http-core/src/test/scala/akka/http/impl/engine/ws/WebSocketIntegrationSpec.scala

```diff
@@ -194,6 +194,40 @@ class WebSocketIntegrationSpec extends AkkaSpec("akka.stream.materializer.debug.
       binding.unbind()
     }
 
+    "terminate the handler flow with an error when the connection is aborted" in Utils.assertAllStagesStopped {
+      val handlerTermination = Promise[Done]
+
+      val handler = Flow[Message]
+        .watchTermination()(Keep.right)
+        .mapMaterializedValue(handlerTermination.completeWith(_))
+        .map(m => TextMessage.Strict(s"Echo [$m]"))
+
+      val bindingFuture = Http().bindAndHandleSync(
+        _.header[UpgradeToWebSocket].get.handleMessages(handler, None),
+        interface = "localhost",
+        port = 0)
+      val binding = Await.result(bindingFuture, 3.seconds.dilated)
+      val myPort = binding.localAddress.getPort
+
+      val ((switch, connection), completion) =
+        Source.maybe
+          .viaMat {
+            Http().webSocketClientLayer(WebSocketRequest("ws://localhost:" + myPort))
+              .atop(TLSPlacebo())
+              .atopMat(KillSwitches.singleBidi[ByteString, ByteString])(Keep.right)
```

Comment:
I guess that could be using a `ClientTransport` instead for hooking in the kill switch but I guess that's ok as well.